### PR TITLE
fix auth: property of int or date type can't be filtered

### PR DESCRIPTION
**Pull request by javeme**
Change-Id: I794165b89b9d21f31938f09daa2041881ba2eb8d

### CODE REVIEW

File: hugegraph-api/src/main/java/com/baidu/hugegraph/auth/HugeAuthenticator.java [hunk 1]
```diff
@@ -48,7 +48,7 @@
     public static final String KEY_PASSWORD =
                                CredentialGraphTokens.PROPERTY_PASSWORD;
     public static final String KEY_ROLE = "role";
-    public static final String KEY_CLIENT = "client";
+    public static final String KEY_CLIENT = "address";
```
Comment (zhoney) [inline hugegraph-api/src/main/java/com/baidu/hugegraph/auth/HugeAuthenticator.java:51]: rename to KEY_ADDRESS

File: hugegraph-api/src/main/java/com/baidu/hugegraph/auth/WsAndHttpBasicAuthHandler.java [hunk 2]
```diff
@@ -0,0 +1,162 @@
...
+            ctx.fireChannelRead(obj);
+        } else {
+            super.channelRead(ctx, obj);
+        }
+    }
+
+    public static boolean isWebSocket(final HttpMessage msg) {
+        final String connectionHeader = msg.headers().get(CONNECTION);
+        final String upgradeHeader = msg.headers().get(UPGRADE);
+        return "Upgrade".equalsIgnoreCase(connectionHeader) ||
```
Comment (zhoney) [inline hugegraph-api/src/main/java/com/baidu/hugegraph/auth/WsAndHttpBasicAuthHandler.java:86]: recheck to make sure it's expected

File: hugegraph-core/src/main/java/com/baidu/hugegraph/schema/PropertyKey.java [hunk 3]
```diff
@@ -214,11 +181,43 @@ public String clazz() {
...
+        }
+        return valid;
+    }
+
     public <V> Object serialValue(V value) {
         V validValue = this.validValue(value);
         E.checkArgument(validValue != null,
                         "Invalid property value '%s' for key '%s'",
                         value, this.name());
+        E.checkArgument(this.cardinality == Cardinality.SINGLE,
```
Comment (zhoney) [inline hugegraph-core/src/main/java/com/baidu/hugegraph/schema/PropertyKey.java:218]: this. cardinality.single()

File: hugegraph-core/src/main/java/com/baidu/hugegraph/schema/PropertyKey.java [hunk 4]
```diff
@@ -238,46 +237,61 @@ public String clazz() {
...
+        if (this.checkValueType(value)) {
+            // Same as expected type, no conversion required
+            return value;
+        }
 
-        V validValue;
+        V validValue = null;
         Collection<T> validValues;
-        if (!(value instanceof Collection)) {
+        if (this.cardinality == Cardinality.SINGLE) {
```
Comment (zhoney) [inline hugegraph-core/src/main/java/com/baidu/hugegraph/schema/PropertyKey.java:261]: this.cardinality.single()

File: hugegraph-api/src/main/java/com/baidu/hugegraph/auth/WsAndHttpBasicAuthHandler.java (hunk 2 above)
Comment (javeme) [reply hugegraph-api/src/main/java/com/baidu/hugegraph/auth/WsAndHttpBasicAuthHandler.java:86]: sure

**Pull request merged**